### PR TITLE
[DinoMod] Implements rideable zombie T Rex via missions

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary

SUMMARY: Mods "Implements rideable zombie T Rex via missions"

#### Purpose of change

Weave together DinoMod content into a clearer progression for the player, add interest and diversity, new mid/late game goal

#### Describe the solution

Two new rideable zombie dinos. Two new Old Guard missions to offer them.

#### Describe alternatives you've considered

New content in the shelter computer, but doesn't appear to be moddable. Dino rescue mission for swampers, couldn't get working, not sure if it's possible

#### Testing

Worldgen works, game can start, and Red NPC spawns and offers first mission. In the last test Betty spawned after completing mission 3 and immediately started eating the Old Guard. Set aggression to 0 to fix.

#### Additional context

Shout out to Mutaman for the T Rex support

### CODE REVIEW

**Comment by LyleSY** [comment]
Travis pit digging weariness test is not related

**Comment by LyleSY** [comment]
I'm still getting circular dependency errors at load, need to strip more back to find the issue

File: data/mods/DinoMod/NPC/NC_Red.json [hunk 1]
```diff
@@ -41,17 +41,17 @@
...
-        "condition": { "not": "npc_following" }
-      },
       { "text": "Let's trade items.", "topic": "TALK_OLD_GUARD_RED", "effect": "start_trade" },
       {
         "text": "How can I help?",
         "topic": "TALK_MISSION_LIST",
         "condition": { "u_has_var": "asked_about_kill_dinos", "type": "dialogue", "context": "dino", "value": "yes" }
       },
+      {
+        "text": "Mission update for you Red.",
```
Comment (actual-nh) [inline data/mods/DinoMod/NPC/NC_Red.json:51]:
```suggestion
        "text": "Mission update for you, Red.",
```

**Comment by LyleSY** [comment]
Stripping out bugfix content for a separate PR

**Comment by LyleSY** [comment]
Sigh, reward dinos eat the NPCs. Needs some more work

**Comment by actual-nh** [comment]
> Sigh, reward dinos eat the NPCs. Needs some more work

There are a few choices:
- Move the reward dinos to a new monster faction (base_faction their original monster faction) that's NEUTRAL or FRIENDLY to the human monster faction. This does mean they'll never attack **any** human.
- Move the reward dinos to a new monster faction that's FRIENDLY to "player", if the NPCs are in the player faction.
- Move the reward dinos to a new monster faction that's FRIENDLY to "player" and to a new monster faction (base_faction human) that the mon_faction entry on the NPCs' faction points to.
- Drop the aggression, but note that anything raising their aggression sufficiently will still result in attacking anything they have BY_MOOD toward (such as humans by default).

**Comment by ZhilkinSerg** [comment]
> 
> 
> Sigh, reward dinos eat the NPCs.

Seems to be working as intended.

**Comment by Maleclypse** [comment]
@LyleSY would this be ready for merge to the dev branch? Or are you still working on the monster faction stuff to keep them from eating NPCs?

**Comment by LyleSY** [comment]
Yes, thanks!

**Review by Maleclypse — APPROVED** [review]

**Pull request merged**